### DISCUSSION TITLE
About setting R version for R Markdown render

### DISCUSSION THREAD

**Discussion by cderv** [REditorSupport/vscode-R#1185] [Q&A] [answered]
Hi,

I have several R versions on my system. 

I just notice that Open R terminal in VSCODE will correctly use R 4.2.1 through _radian_. 
However, render a R Markdown document will not use this version of R but another one. 

At first I did not understand why, but after a few dig and tries into the settings I noticed: 

R Terminal is controled by config in R extension
> R › Rterm: Windows
> R path for Windows. Can also be radian etc.

but the render button on R Markdown document seems to be controlled by config in R Debugger
> R › Rpath: Windows
> Path to an R executable for Windows. Must be "vanilla" R, not radian etc.!

Is this expected ? 

How can I be sure the two are in sync if I am using `radian` for R Terminal ? Is there a good way ? 

For now I just deactivated my setting in R Debugger. I passed no path to R, and it seemed to work and take my system default one, which should be the same as in Radian. 

Just sharing here as a question as I did not found a clear answer to this in the configuration doc. I probably should not have set up RDebugger path anyway, I don't remember why indeed. 

Hope it helps see if this is a problem, or just a misunderstanding from me reading the documentation. 

Thanks for all the hard work on this extension !

**Comment by andycraig** [comment, marked as answer]
Both the `r.rterm` and `r.rpath` settings come from this extension (not R Debugger). The `r.rterm` setting is used to decide what console will be started (e.g., radian), and the `r.rpath` setting is used for everything else (e.g., R Markdown, help info).

To make sure the same version of R is used you may need to specify which version of R is used by radian: https://github.com/randy3k/radian#how-to-switch-to-a-different-r-or-specify-the-version-of-r

**Reply by cderv** [reply]
Thank you for the answer. So two configurations are required and need to stay in thing. I'll adapt to that. I wasn't sure so thanks for taking the time to confirm. 

> Both the r.rterm and r.rpath settings come from this extension (not R Debugger). 

In the Setting Pane inside VSCODE, RPath is under the R Debugger extensions so I was mislead by this

<details><summary>Screenshot</summary>

![image](https://user-images.githubusercontent.com/6791940/187625015-2bb86a2b-89ae-4ae6-aff1-b39053a2adeb.png)
</details>

**Reply by andycraig** [reply]
No worries, glad I could help!

VS Code's layout of the R and R Debugger settings is odd - I wasn't aware of that. It's good to know. Thank you for the screenshot.
